comp/metadata/runner/runnerimpl/runner.go

```diff
@@ -137,12 +176,21 @@ func (r *runnerImpl) handleProvider(p MetadataProvider) {
 // start is called by FX when the application starts. Lifecycle hooks are blocking and called sequencially. We should
 // not block here.
 func (r *runnerImpl) start() error {
-	r.log.Debugf("Starting metadata runner with %d providers", len(r.providers))
-	for _, optionaP := range r.providers {
-		if p, isSet := optionaP.Get(); isSet {
-			go r.handleProvider(p)
+	r.log.Debugf("Starting metadata runner with %d providers", len(r.providers)+len(r.priorityProviders))
```

Comment:
Having the split between both type might help troubleshooting.
```suggestion
	r.log.Debugf("Starting metadata runner with %d priority providers and %d regular providers", len(r.providers), len(r.priorityProviders))
```